File: project-service/Dtos/Tasks/TaskResponse.cs
namespace ProjectService.Dtos.Tasks;

public class TaskResponse
{
    public Guid TaskId { get; set; }
    public Guid ProjectId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public Guid? AssigneeId { get; set; }
    public string Status { get; set; } = string.Empty;
    public decimal EstimateHours { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public TaskProjectSummary? Project { get; set; }
}

public class TaskProjectSummary
{
    public Guid ProjectId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
}

File: time-logging-service/src/main/java/com/automobileservice/time_logging_service/client/dto/TaskResponse.java
package com.automobileservice.time_logging_service.client.dto;

import lombok.Data;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
public class TaskResponse {
    private UUID taskId;
    private UUID projectId;
    private String title;
    private String serviceType;
    private String detail;
    private String status;
    private Long assigneeId;
    private Double estimateHours;
    private OffsetDateTime scheduledStart;
    private OffsetDateTime scheduledEnd;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}
